authentication/views/authentication.py

```diff
@@ -0,0 +1,73 @@
+from datetime import datetime
+from django.conf import settings
+import jwt
+from rest_framework import status, views
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.response import Response
+
+from authentication.token import decode_token
+
+
+class LoginView(views.APIView):
+    """This view takes a JWT as a GET parameter, verifies it and sets it as a cookie."""
```

Comment:
There should be some kind of check such as a state parameter to make sure the JWT comes from the SSO server. Otherwise this could be a vector for an attack
https://auth0.com/docs/protocols/oauth2/oauth-state

Reply:
There is an `iss` claim + the RSA signing, isn't that enough?

Reply:
(which are all checked for in `decode_token`)

Reply:
The state will prevent CSRF attacks, it has nothing to do with encryption. Currently someone could send the user a link and set arbitrary cookies in their browser

Reply:
Correct me if I'm wrong, but I read the code once again, and I see this logic:
if the GET parameter is not correctly decoded by `decode_token`, raise an Exception (and thus don't set the cookie)
So I believe an arbitrary string (and actually anything that is not a valid token) will not be set as a cookie